### PR TITLE
Prevent timeout failure

### PR DESCRIPTION
**Pull request by dimadio**
Prevent timeout failure …
At this point RecvFrom is undefined, replaced with From.
Steps to reproduce - call handshake_continue with very low timeout (1) and see crash report with
exception error: {bad_reply_action_from_state_function,
{reply,undefined,{error,timeout}}}

### CODE REVIEW

**Comment by IngelaAndin** [comment]
Thank you for making the required changes, alas it just missed the deadline for OTP 21.2, but there, of course, will be more chances and we will include it for testing again as soon as possible.

File: lib/ssl/src/ssl_connection.erl [hunk 1]
```diff
@@ -668,9 +668,11 @@ user_hello({call, From}, cancel, #state{negotiated_version = Version} = State, _
                      Version, ?FUNCTION_NAME, State);
 user_hello({call, From}, {handshake_continue, NewOptions, Timeout}, #state{hello = Hello,
                                                                            role = Role,
-                                                                           start_or_recv_from = RecvFrom,
+                                                                           start_or_recv_from = _RecvFrom,
```
Comment (IngelaAndin) [inline lib/ssl/src/ssl_connection.erl]: You can remove this line as it no longer needed!

Comment (dimadio) [reply lib/ssl/src/ssl_connection.erl]: Removed

**Comment by IngelaAndin** [comment]
Looking good now, but you do not have to skip the test for DTLS it should work for DTLS too. This functionality is only handle by shared code between TLS and DTLS.  DTLS is specified as a diff  against TLS and typically tls_* modules handles TLS specifics, dtls_* modules handles DTLS specifics and ssl_* modules are shared.

**Comment by dimadio** [comment]
For some reason, probably due to my non-existing knowledge of ssl_test_lib, all DTLS tests fail with timetrap_timeout for me. Please help.

**Comment by dimadio** [comment]
I believe I found the problem. Since DTLS uses UDP sockets it is unaware of timed out server socket. ssl_test_lib:client_loop calls to connect method without timeout property so in this case, UDP sockets hang. I add an ability to pass a timeout to options (defaults to infinity) and it doesn't bring damage and solves the issue. I will push the test fix after additional checks.

**Comment by IngelaAndin** [comment]
The test has now been running for a while and I think that the test case has the problem that
it can either return {error, timeout} or {error, closed} for the server depending on timing.  I do not think
it is necessary to have a timeout  on the client side for this test case. It could be interesting to have another test case for the client timeout though.

**Comment by dimadio** [comment]
> The test has now been running for a while and I think that the test case has the problem that
> it can either return {error, timeout} or {error, closed} for the server depending on timing. I do not think
> it is necessary to have a timeout on the client side for this test case. It could be interesting to have another test case for the client timeout though.

It's necessary to have the timeout, otherwise on server socket close UDP based (DTLS)  connection is not closed.

**Comment by IngelaAndin** [comment]
Due to many other TLS activities it has been hard to follow up on the test for this specif branch, but things are clearing up so I hope to have a result soon. Hang in there.

**Comment by IngelaAndin** [comment]
I am currently working on optimizations of the TLS  code base and as it happens I rewrote the handling
of these timers to use a new gen_statem feature. So I cherry-picked your test case from this PR to make sure that I do not reintroduce the bug, however the code change is no longer valid so I will close this PR.  Fix should show up on maint and master branch soon, but it includes a lot of other changes so it needs to brew in our tests for a couple of days more.
